### ISSUE TITLE
Support for TorchSnapshot for efficient checkpoint saving and loading

### ISSUE THREAD

**Issue by ananthsub** [pytorch/ignite#2752]
## 🚀 Feature

TorchSnapshot is a performant, memory-efficient checkpointing library for PyTorch applications, designed with large, complex distributed workloads in mind. It includes many optimizations to control for memory usage and optimize checkpoint writing for DDP-style workloads over torch.save/torch.load. For more information, please check out the readme: https://github.com/pytorch/torchsnapshot#why-torchsnapshot

This could be a nice addition to Ignite, similar to the existing Checkpoint handler

cc @yifuwang

**Comment by ananthsub** [comment]
> A question I have about the usage, in DDP user should call Snapshot.take by all ranks ?

Yes, Snapshot.take should always be called on all ranks in a distributed setting. It acts as a collective.

> How about the path specified in the argument, where it should be, node 0, rank 0 ?

The path specified should be a directory, which should be the same across all ranks. If on a multi-node setting, this assumes you have a storage system visible by all nodes (e.g. a cloud storage object store)

**Comment by aniketpandey05** [comment]
Hi @vfdev-5 , is this still relevant? I went through checkpoint.py and was thinking a TorchSnapshotSaver(BaseSaveHandler) wrapping Snapshot.take in __call__ and handling directory removal in remove() might work . Since Snapshot.take is collective in DDP, no rankgating would be needed unlike DiskSaver.

**Comment by aliceyh7** [comment]
Hi! I'm a student in the CodePath AI301 program and I'd like to work on this issue. I'll start by reproducing the problem locally this week. Any pointers on where to start in the codebase would be appreciated!
